### PR TITLE
feat(config): add trusted project config discovery

### LINKED ISSUE #239: Feature request: Per project configuration
While restish works great for personal use, it would be even better if you could easily share an API configuration in a project. The use case is api-specific commands in scripts, improving readability. That would require a way for restish to automatically pick up local configuration files, or to specify a file on the command line.

It seems like relative paths to spec_files already works, so hopefully this would play nicely with this feature.

### PR DESCRIPTION
**Pull request by danielgtaylor**
## Summary

- Adds trusted `.restish.json` discovery from the current directory or parents.
- Adds `restish config trust` with path+content-hash trust stored outside the repo.
- Layers trusted project `apis` and `theme` over global config while keeping project APIs read-only for normal mutation commands.
- Namespaces project API spec cache, HTTP cache, and API-scoped OAuth token cache outside the repo.
- Updates design docs and user docs for the new trust model.

Fixes #239.

Related to #263 by @tarunKoyalwar and #303 by @richard-hajek. Thanks to both original authors for exploring the project-config workflow and edge cases.

## Validation

- `env GOCACHE=/tmp/restish-gocache go test ./internal/cli -run TestProjectConfig`
- `env GOCACHE=/tmp/restish-gocache go test ./internal/cli`
- `env GOCACHE=/tmp/restish-gocache go test ./...`
- `env GOCACHE=/tmp/restish-gocache go run ./cmd/restish-docgen --check`
- `hugo --source site --quiet --gc --minify --cacheDir /tmp/restish-hugo-cache`
- `git diff --check`
- `env GOCACHE=/tmp/restish-gocache go test -tags=integration ./...`

## Review

Ran a no-context sub-agent review. It found three issues: project config could leak into global config on `api remove`, trusted project APIs were skipped for help/completion bootstrap commands, and project APIs could not reference global auth profiles. All three were fixed with targeted tests.

### CODE REVIEW

File: internal/cli/project_config.go [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+	return " (" + strings.Join(parts, "; ") + ")", nil
+}
+
+func (c *CLI) projectConfigRuntimeSubset(project *projectConfigState) (*config.Config, map[string]bool, error) {
+	if project == nil {
+		return nil, nil, nil
+	}
+	if insecure, permErr := config.ConfigFileHasInsecurePermissions(project.Path); permErr == nil && insecure {
+		return nil, nil, fmt.Errorf("%s is group/world-readable; project config can contain credentials (chmod 600)", project.Path)
+	}
```
[the remark this answers is not in the document]
Comment (richard-hajek) [reply internal/cli/project_config.go]: I don't think the chmod 600 is necessary. Really the only project I've seen do this is OpenSSH, which enforces permissions on the keys. Not even various dotenv management projects enforce this

**Comment by danielgtaylor** [comment]
Follow-up from the project-config decision:

- `.restish.json` is intended to be a committed, shareable project file, so the implementation now allows normal repository permissions such as `0644`.
- The safety boundary is content validation plus explicit trust: project config rejects inline secret-bearing auth params and credential-bearing persistent headers/query params.
- Secret auth params can be omitted or written as `env:NAME` references. Non-secret OAuth values like `client_id`, `audience`, scopes, issuer URLs, and token URLs remain valid shared setup.
- I also carried forward the PR #340-only migration and implementation-contract doc notes, so closing #340 should not drop useful design text.

Validation rerun after this change:

- `env GOCACHE=/tmp/restish-gocache go test ./internal/cli -run TestProjectConfig`
- `env GOCACHE=/tmp/restish-gocache go test ./internal/cli`
- `env GOCACHE=/tmp/restish-gocache go test ./...`
- `env GOCACHE=/tmp/restish-gocache go run ./cmd/restish-docgen --check`
- `hugo --source site --quiet --gc --minify --cacheDir /tmp/restish-hugo-cache`
- `git diff --check`

File: internal/cli/project_config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply internal/cli/project_config.go]: Addressed in `1e981201`: auto-discovered `.restish.json` no longer requires private mode bits, so a normal committed `0644` file works. The safety boundary is now content validation: inline secret auth params and credential-bearing persistent headers/query params are rejected, while secret auth params can use `env:NAME` references and non-secret OAuth params remain allowed.

File: internal/cli/project_config.go [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+func projectConfigHasUnsupportedTopLevel(cfg *config.Config) bool {
+	if cfg == nil {
+		return false
+	}
+	return len(cfg.AuthProfiles) > 0 ||
+		cfg.Cache != (config.CacheConfig{}) ||
+		len(cfg.Plugins) > 0 ||
+		cfg.ThemeSource != ""
+}
```
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply internal/cli/project_config.go]: Addressed in `943fc3b8`: project config is decoded through raw top-level keys first, so unsupported keys are rejected by presence even when their values are empty. Covered by `TestProjectConfigRejectsEmptyUnsupportedTopLevelKeys`.

File: site/content/en/docs/reference/config.md [hunk 3]
```diff
@@ -41,9 +41,24 @@ If no default config directory can be determined, set `RSH_CONFIG` or
...
+Trust is stored outside the repository and includes the file's content hash. If
+`.restish.json` changes, run `restish config trust` again after reviewing it.
+Trusted project config layers `apis` and `theme` over your global config. Project
+APIs override global APIs with the same name, while unrelated global APIs remain
+available. Normal config-writing commands still write the global config and
+refuse to mutate APIs that came from trusted project config.
+
+Use `--rsh-config ./restish.json` or `RSH_CONFIG=./restish.json` when a command
+should use one complete project config file instead of layering trusted project
+config over the global config.
```
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply site/content/en/docs/reference/config.md]: Addressed in `943fc3b8`: the explicit-config example now uses `.restish.json`, and the docs call out that explicit config selection can point to any filename while `.restish.json` is the auto-discovered name.

File: internal/cli/config_cmd.go [hunk 4]
```diff
@@ -101,7 +109,11 @@ func (c *CLI) runConfigShow(cmd *cobra.Command, args []string) error {
 		cfg = &config.Config{}
 	}
 	style := humanTextStyleFor(c.Stdout)
-	fmt.Fprintf(c.Stdout, "%s %s\n", style.key("Config file:"), c.configFilePath())
+	fmt.Fprintf(c.Stdout, "%s %s\n", style.key("Config file:"), c.configSourceSummary())
+	if c.projectConfig != nil && c.projectConfig.Trusted {
+		names := sortedProjectAPINames(c.projectConfig.APIs)
+		fmt.Fprintf(c.Stdout, "%s %s (%s)\n", style.key("Project config:"), c.projectConfig.Path, strings.Join(names, ", "))
```
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply internal/cli/config_cmd.go]: Addressed in `943fc3b8`: `config show` now omits the empty API-name parentheses when trusted project config only provides theme overrides. Covered by `TestProjectConfigShowOmitsEmptyAPIList`.

File: internal/cli/project_config.go [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+
+func (c *CLI) validateProjectAuthNoInlineSecrets(path string, authCfg *config.AuthConfig) error {
+	if authCfg == nil || authCfg.Type == "" {
+		return nil
+	}
+	handler, err := c.authHandlerFor(authCfg, authHandlerOptions{})
+	if err != nil {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply internal/cli/project_config.go:388]: Addressed in `f969a828`: project config validation now returns the `authHandlerFor` error for unknown auth types instead of silently accepting the block. `TestProjectConfigRejectsUnknownAuthType` covers `config trust` failing with the auth path in the diagnostic.

File: internal/cli/api.go [hunk 6]
```diff
@@ -250,7 +250,7 @@ func (c *CLI) runAPISync(cmd *cobra.Command, args []string) error {
 		if err := c.configureAllowedOperationOrigins(cmd, apiName, syncedCfg, apiSpec, profileName, yes); err != nil {
 			return err
 		}
-		if !reflect.DeepEqual(apiCfg, syncedCfg) {
+		if !reflect.DeepEqual(apiCfg, syncedCfg) && !c.projectAPI(apiName) {
 			if err := c.saveAPIConfig("api sync", apiName, c.cfg, syncedCfg); err != nil {
```
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply internal/cli/api.go:254]: Addressed in `f969a828`: `configureAllowedOperationOrigins` now detects read-only project APIs before prompting or applying `--yes`. It emits a project-config-specific warning that points at editing `.restish.json` or using `--rsh-config`, and avoids suggesting `api set`. Covered by `TestProjectConfigAPISyncWarnsWhenOperationOriginsNeedProjectEdit`.

File: internal/cli/project_config.go [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+		return nil, fmt.Errorf("config: cannot read %s: %w", path, err)
+	}
+	stripped := jsonc.ToJSON(data)
+	var raw map[string]json.RawMessage
+	if err := decodeProjectConfigJSON(path, stripped, &raw); err != nil {
+		return nil, err
+	}
+	for key := range raw {
+		if key != "apis" && key != "theme" {
+			return nil, fmt.Errorf("project config: only apis and theme are supported in %s for now", projectConfigFileName)
```
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply internal/cli/project_config.go]: Addressed in `2db95000`: unsupported top-level project config keys now report the concrete config path and rejected key name, with deterministic key selection. `TestProjectConfigRejectsEmptyUnsupportedTopLevelKeys` covers the diagnostic.

File: internal/cli/cache_cmd.go [hunk 8]
```diff
@@ -273,7 +273,11 @@ func (c *CLI) newCacheClearCmd() *cobra.Command {
 			if len(args) == 1 {
 				apiName := args[0]
 				registered := c.cfg != nil && c.cfg.APIs != nil && c.cfg.APIs[apiName] != nil
-				cleared, err := dc.ClearNamespacePrefix(apiName + ":")
+				namespace := apiName
+				if registered {
+					namespace = c.apiStateName(apiName)
+				}
+				cleared, err := dc.ClearNamespacePrefix(namespace + ":")
```
[the remark this answers is not in the document]
Comment (danielgtaylor) [reply internal/cli/cache_cmd.go:295]: Addressed in `2db95000`: cache info, JSON output, and the terminal treemap now map trusted project cache namespaces back to the logical API/profile before deciding whether they are registered. `TestCacheInfoShowsProjectAPINamespaceAsRegistered` covers the human and JSON output.
